libraries/chain/controller.cpp

```diff
@@ -3632,7 +3633,7 @@ struct controller_impl {
          ilog("Received block ${id}... #${n} @ ${t} signed by ${p} " // "Received" instead of "Applied" so it matches existing log output
               "[trxs: ${count}, lib: ${lib}, net: ${net}, cpu: ${cpu}, elapsed: ${elapsed}, time: ${time}, latency: ${latency} ms]",
               ("p", bsp->producer())("id", bsp->id().str().substr(8, 16))("n", bsp->block_num())("t", bsp->timestamp())
-              ("count", bsp->block->transactions.size())("lib", fork_db_root_block_num())
+              ("count", bsp->block->transactions.size())("lib", bsp->irreversible_blocknum())
```

Comment:
Should be `chain_head.irreversible_blocknum()` I think, otherwise if the block was not applied it may claim a lib higher than the next log message at line 3645.

Reply:
All the info here is in relation to `bsp`. If the block was not applied to head then you want the different values. The point of `Block not applied to head` log is so you can see the difference between the received block and the chain_head.

Reply:
Sure seems weird that `lib` would go *backwards* from one log message to the next, but OK.
Especially since in the case of proper blocks the lib really has advanced I think.